File: 1-KMP-PREFIX/PYTHON/1-BRUTE-FORCE/go/main.go
package main

import "fmt"

func bruteForce(s string) []int {
	n := len(s)
	pi := make([]int, n)

	for i := 0; i < n; i++ {
		for L := i; L > 0; L-- {
			if s[:L] == s[i-L+1:i+1] {
				pi[i] = L
				break
			}
		}
	}
	return pi
}

func main() {
	var s string
	fmt.Scan(&s)
	a := bruteForce(s)
	fmt.Println(a)
}

File: 1-KMP-PREFIX/PYTHON/1-BRUTE-FORCE/python/main.py
s=input()
def brute_force(s):
    n=len(s)    
    pi=[0]*n
    for i in range(n):
        for L in range(i,0,-1):
            if(s[:L]==s[i-L+1:i+1]):
                pi[i]=L
                break
    return pi

a=brute_force(s)
print(a)

